src/Collapse.tsx

```diff
@@ -229,19 +230,20 @@ const Collapse = React.forwardRef(
         onEntered={handleEntered}
         onExit={handleExit}
         onExiting={handleExiting}
+        childRef={(children as any).ref}
```

Comment:
this will always be falsey, children, if it's an element would have ref on `children.props` not on the element itself

Reply:
Hmm `children.ref` is actually defined when I was testing this.  I've tried `children.props.ref` but I get a warning:

Warning: div: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop.

Reply:
tbh, children.ref did seem kinda hacky (esp. with the casting).  I'm not 100% sure on what the correct approach is :P

Reply:
o hmm i must have it backwards, been doing this for too long. I'm suprised the types don't correctly cover this tho

Reply:
I found this on the react github project that seems to confirm the `children.ref` usage:

https://github.com/facebook/react/issues/8873#issuecomment-275423780